central/auth/datastore/datastore_impl_test.go

```diff
@@ -0,0 +1,125 @@
+//go:build sql_integration
+
+package datastore
+
+import (
+	"context"
+	"testing"
+
+	pgStore "github.com/stackrox/rox/central/auth/store/postgres"
+	roleDataStore "github.com/stackrox/rox/central/role/datastore"
+	permissionSetPostgresStore "github.com/stackrox/rox/central/role/store/permissionset/postgres"
+	rolePostgresStore "github.com/stackrox/rox/central/role/store/role/postgres"
+	accessScopePostgresStore "github.com/stackrox/rox/central/role/store/simpleaccessscope/postgres"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/errox"
+	"github.com/stackrox/rox/pkg/postgres/pgtest"
+	"github.com/stackrox/rox/pkg/sac"
+	"github.com/stackrox/rox/pkg/sac/resources"
+	"github.com/stackrox/rox/pkg/set"
+	"github.com/stackrox/rox/pkg/uuid"
+	"github.com/stretchr/testify/suite"
+)
+
+const (
+	testRole1 = "New-Admin"
+	testRole2 = "Super-Admin"
+	testRole3 = "Super Continuous Integration"
+)
+
+var (
+	existingRoles = set.NewFrozenStringSet(testRole1, testRole2, testRole3)
+)
+
+func TestAuthDatastorePostgres(t *testing.T) {
+	suite.Run(t, new(datastorePostgresTestSuite))
+}
+
+type datastorePostgresTestSuite struct {
+	suite.Suite
+
+	ctx           context.Context
+	pool          *pgtest.TestPostgres
+	authDataStore DataStore
+	roleDataStore roleDataStore.DataStore
+}
+
+func (s *datastorePostgresTestSuite) SetupTest() {
+	s.ctx = sac.WithGlobalAccessScopeChecker(context.Background(),
+		sac.AllowFixedScopes(
+			sac.AccessModeScopeKeys(storage.Access_READ_ACCESS, storage.Access_READ_WRITE_ACCESS),
+			sac.ResourceScopeKeys(resources.Access),
+		),
+	)
+
+	s.pool = pgtest.ForT(s.T())
+	s.Require().NotNil(s.pool)
+
+	store := pgStore.New(s.pool.DB)
+	s.authDataStore = New(store)
+
+	permSetStore := permissionSetPostgresStore.New(s.pool.DB)
+	accessScopeStore := accessScopePostgresStore.New(s.pool.DB)
+	roleStore := rolePostgresStore.New(s.pool.DB)
+	s.roleDataStore = roleDataStore.New(roleStore, permSetStore, accessScopeStore, func(_ context.Context, _ func(*storage.Group) bool) ([]*storage.Group, error) {
+		return nil, nil
+	})
+	s.addRoles()
```

Comment:
[minor] Please add a blank line. I overlooked `s.addRoles()` on the first pass.

```suggestion
	})

	s.addRoles()
```

I personally prefer having things like `s.addRoles()` in test functions rather than in the setup even if calls are duplicated. I find test functions more self-describing this way. Not insisting at all, it's your call.